server/eclipse-project/src/main/java/au/org/democracydevelopers/corla/query/GenerateAssertionsSummaryQueries.java

```diff
@@ -0,0 +1,106 @@
+version 3 of the License, or (at your option) any later version.
+
+The colorado-rla IRV extensions are distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+See the GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along with
+raire-service. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+package au.org.democracydevelopers.corla.query;
+
+import au.org.democracydevelopers.corla.model.GenerateAssertionsSummary;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.hibernate.Session;
+import us.freeandfair.corla.persistence.Persistence;
+
+import javax.persistence.TypedQuery;
+import java.util.List;
+import java.util.Optional;
+
+import static au.org.democracydevelopers.corla.endpoint.GenerateAssertions.UNKNOWN_WINNER;
+
+/**
+ * Database queries relating to the retrieval of Assertion Generation Summaries from the
+ * database. It contains a method that executes a query to retrieval all GenerateAssertionsSummaries
+ * belonging to a specific contest, identified by name.
+ * Also includes a shortcut function to get the winner, which inserts UNKNOWN_WINNER if either the
+ * record is absent, or the winner is blank.
+ * (e.g. giving feedback about assertion generation) they can be safely deleted.
+ */
+public class GenerateAssertionsSummaryQueries {
+
+  /**
+   * Class-wide logger.
+   */
+  public static final Logger LOGGER = LogManager.getLogger(GenerateAssertionsSummaryQueries.class);
+
+  /**
+   * Retrieve the winner of an IRV contest matching the given contestName, or UNKNOWN_WINNER if
+   * there is no record, or no winner in the record.
+   * @param contestName the name of the contest.
+   */
+  public static String matchingWinner(final String contestName) {
+    Optional<GenerateAssertionsSummary> optSummary = matching(contestName);
+    if(optSummary.isPresent() && !optSummary.get().winner.isBlank()) {
+      return optSummary.get().winner;
+    } else {
+      return UNKNOWN_WINNER;
+    }
+  }
+
+  /**
+   * Retrieve all summaries in the database belonging to the contest with the given name.
+   * @param contestName The contest name.
+   * @return the (optional) summary of assertions defined for the contest.
+   * @throws RuntimeException when an unexpected error arose in assertion retrieval (not including
+   * a NoResultException, which is handled by returning an empty optional item).
+   */
+  public static Optional<GenerateAssertionsSummary> matching(final String contestName) throws RuntimeException {
+    final String prefix = "[matching]";
+    try {
+      LOGGER.debug(String.format("%s Select query on generate assertions summary table, retrieving " +
+          "summary for contest with name %s.", prefix, contestName));
+
```

Comment:
colorado-rla seems to put queries in the "queries" package. I don't know if we want to push the query construction and execution  into AssertionQueries.java?